shell/browser/api/electron_api_menu_mac.mm

```diff
@@ -164,19 +164,37 @@
           static_cast<content::RenderWidgetHostViewMac*>(rfh->GetView());
       RenderWidgetHostViewCocoa* cocoa_view = rwhvm->GetInProcessNSView();
       view = cocoa_view;
+
+      // TODO: ui::ShowContextMenu does not dispatch the event correctly
+      // if no frame is found. Fix this to remove if/else condition.
+      NSEvent* dummy_event =
+          [NSEvent mouseEventWithType:NSEventTypeRightMouseDown
+                             location:position
+                        modifierFlags:0
+                            timestamp:0
+                         windowNumber:nswindow.windowNumber
+                              context:nil
+                          eventNumber:0
+                           clickCount:1
+                             pressure:0];
+      ui::ShowContextMenu(menu, dummy_event, view, true);
     }
+    // If no render frame host was found, return early.
+    return;
+  } else {
+    // Make sure events can be pumped while the menu is up.
+    base::CurrentThread::ScopedAllowApplicationTasksInNativeNestedLoop allow;
+
+    // One of the events that could be pumped is |window.close()|.
+    // User-initiated event-tracking loops protect against this by
+    // setting flags in -[CrApplication sendEvent:], but since
+    // web-content menus are initiated by IPC message the setup has to
+    // be done manually.
+    base::mac::ScopedSendingEvent sendingEventScoper;
+
+    // Don't emit unresponsive event when showing menu.
+    [menu popUpMenuPositioningItem:item atLocation:position inView:view];
   }
```

Comment:
We still want to show a menu when `frame` isn't provided, or if things don't work out in the `frame` case.

```suggestion
      return;
    }
  }

  // Make sure events can be pumped while the menu is up.
  base::CurrentThread::ScopedAllowApplicationTasksInNativeNestedLoop allow;

  // One of the events that could be pumped is |window.close()|.
  // User-initiated event-tracking loops protect against this by
  // setting flags in -[CrApplication sendEvent:], but since
  // web-content menus are initiated by IPC message the setup has to
  // be done manually.
  base::mac::ScopedSendingEvent sendingEventScoper;

  // Don't emit unresponsive event when showing menu.
  [menu popUpMenuPositioningItem:item atLocation:position inView:view];
```

Reply:
Sorry I meant something like this ☝️ 🙇